### PR TITLE
Add more ScalaJs using directives

### PR DESCRIPTION
**Pull request by lwronski**
Add support for configure bellow ScalaJs options in using directives:

- jsHeader
- jsAllowBigIntsForLongs
- jsAvoidClasses
- jsAvoidLetsAndConsts
- jsModuleSplitStyle
- jsEsVersion

### CODE REVIEW

File: modules/build/src/main/scala/scala/build/options/ScalaJsOptions.scala [hunk 1]
```diff
@@ -37,6 +50,26 @@ final case class ScalaJsOptions(
       case _                     => ModuleKind.NoModule
     }
 
+  def moduleSplitStyle: ModuleSplitStyle =
+    moduleSplitStyleStr.map(_.trim.toLowerCase(Locale.ROOT)).getOrElse("") match {
+      case "fewestmodules"   => ModuleSplitStyle.FewestModules
+      case "smallestmodules" => ModuleSplitStyle.SmallestModules
+      case _                 => ModuleSplitStyle.FewestModules
```
Comment (romanowski) [inline modules/build/src/main/scala/scala/build/options/ScalaJsOptions.scala]: Shouldn't it be an error?

File: modules/build/src/main/scala/scala/build/options/ScalaJsOptions.scala [hunk 2]
```diff
@@ -37,6 +50,26 @@ final case class ScalaJsOptions(
...
+    esVersionStr.map(_.trim.toLowerCase(Locale.ROOT)).getOrElse("") match {
+      case "es5_1"  => ESVersion.ES5_1
+      case "es2015" => ESVersion.ES2015
+      case "es2016" => ESVersion.ES2016
+      case "es2017" => ESVersion.ES2017
+      case "es2018" => ESVersion.ES2018
+      case "es2019" => ESVersion.ES2019
+      case "es2020" => ESVersion.ES2020
+      case "es2021" => ESVersion.ES2021
+      case _        => ESFeatures.Defaults.esVersion
```
Comment (romanowski) [inline modules/build/src/main/scala/scala/build/options/ScalaJsOptions.scala]: Shouldn't it be an error?

**Review by romanowski — CHANGES_REQUESTED** [review]
We should update JS guide as well.

File: modules/build/src/main/scala/scala/build/options/ScalaJsOptions.scala (hunk 1 above)
Comment (lwronski) [reply modules/build/src/main/scala/scala/build/options/ScalaJsOptions.scala]: Previously, we used the default version for `moduleKind`, when someone pass incorrect value. 

and now I added a warning logging, is it good enough?

**Review by romanowski — APPROVED** [review]

**Pull request merged**